ddtrace/internal/writer.py

```diff
@@ -76,6 +76,12 @@ def _send_stats(self):
         return bool(config.health_metrics_enabled and self.dogstatsd)
 
     def write(self, spans=None, services=None):
+        # Start the AgentWriter on first write.
+        # Starting it earlier might be an issue with gevent, see:
+        # https://github.com/DataDog/dd-trace-py/issues/1192
+        if self._started is False:
```

Comment:
do we need to think about when the main process forks?

Reply:
If the process forks, the thread does not exist anymore. We then recreate the `AgentWriter` from scratch and will restart-it on `write`. So that should be good.

There's only one case I think where we have an issue, is then `configure` is called several times. The previous thread is never stopped.

Reply:
we are probably fine to ignore that scenario, as long as we handle when the process forks (which is a common case), then probably good to go!

Reply:
do we need to maintain `self.started` and `self._started` separately?

I had tried this with just `s/self.started/self.is_alive()/` which should roughly be the same thing ... ?

Reply:
The trap is that `is_alive` will return `False` is the thread exited. which would mean the writer would try to start it again… which would fail because you can't re-start a thread that stopped.
So I think it's safer this way, we never re-try to start the thread even if it exited for whatever reason.